src/app/user/auth/auth/auth.component.html

```diff
@@ -29,30 +31,31 @@
                     <mat-label>
                         First name
                     </mat-label>
-                    <input name="title" alt="email" matInput type="text" required [(ngModel)]="firstName"/>
+                    <input name="title" alt="email" matInput type="text" required pattern="[a-zA-Z][a-zA-Z]*" #fname="ngModel" [(ngModel)]="firstName"/>
                 </mat-form-field>
                 <mat-form-field hideRequiredMarker color="primary" formnovalidate name="last-name" appearance="outline"
                      >
                     <mat-label>
                         Last name
                     </mat-label>
-                    <input name="title" alt="email" matInput type="text" required [(ngModel)]="lastName"/>
+                    <input name="title" alt="email" matInput type="text" required pattern="[a-zA-Z][a-zA-Z]*" #lname="ngModel" [(ngModel)]="lastName"/>
                 </mat-form-field>
                 <mat-form-field hideRequiredMarker color="primary" formnovalidate name="email" appearance="outline"
                      >
                     <mat-label>
                         Email
                     </mat-label>
-                    <input name="title" alt="email" matInput type="text" required [(ngModel)]="email"/>
+                    <input name="title" alt="email" matInput type="text" required pattern="^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,3})$" #signupEmail = "ngModel" [(ngModel)]="email"/>
                 </mat-form-field>
                 <mat-form-field hideRequiredMarker formnovalidate color="primary" name="password" appearance="outline"
                      >
                     <mat-label>
                         Password
                     </mat-label>
-                    <input name="title" alt="pass" matInput type="password" required [(ngModel)]="password"/>
-                </mat-form-field> <br>
-                <button id="auth-button" mat-button (click)="signUp()">
+                    <input name="title" alt="pass" matInput type="password" required minlength="8" #signupPassword="ngModel" [(ngModel)]="password"/>
+                    <mat-hint align="start"> <strong style="font-size: 12px;">must be at least 8 characters long</strong></mat-hint>
```

Comment:
add red color to it, maybe?

Reply:
no, I think red points to error, hint isn't an error

Reply:
okay